### ISSUE TITLE
Declaration files don't work with multiple entrypoints

### ISSUE THREAD

**Issue by kristojorg** [egoist/tsup#316]
This is my `tsup` config:

```json
"tsup": {
    "splitting": false,
    "sourcemap": true,
    "clean": true,
    "dts": true,
    "format": [
      "esm",
      "cjs"
    ],
    "entryPoints": [
      "src/index.ts",
      "src/cli/index.ts"
    ]
  },
```

When I run build, the file `dist/index.d.ts` only has `#!/usr/bin/env node` as contents (even though this is not the CLI declaration file), nothing else. There is no file `dist/cli/index.d.ts`. If I run it with only a single entrypoint: `src/index.ts`, the `dist/index.d.ts` file is generated properly to correspond to `src/index.ts`.

**Comment by kristojorg** [comment]
Found a workaround:
```
dts: {
    entry: {
      index: "src/index.ts",
      "src/cli/index": "src/cli/index.ts",
    },
  },
```

This will create a `dist/cli/index.d.ts` file as well as a `dist/index.d.ts` file

**Comment by cyberalien** [comment]
I've ran into the same issue. It does generate `index.d.ts` for both files, but it puts them all in output directory ignoring subdirectories, thus in your example it is creating the same file twice.

So issue is `dts` option ignores sub-directories, dumping all generated files in `outDir`.

**Comment by egoist** [comment]
@sachinraja should be fixed now in v5.4.4

**Comment by sachinraja** [comment]
That fixed it, thanks!
